mempool/cat/store.go

```diff
@@ -58,6 +63,9 @@ func (s *store) remove(txKey types.TxKey) bool {
 		return false
 	}
 	s.bytes -= tx.size()
+	if err := s.deleteOrderedTx(tx); err != nil {
+		panic(err)
```

Comment:
I'm concerned about the panic here. Should we consider:
1. logging this error and `return false` instead? 
2. modifying the `remove` function signature to propagate the error?

Reply:
I can switch to logging the error. The panic is in place because this would imply a developer error i.e. this shouldn't error so long as the developer has correctly implemented this method (the tx should always be in the ordered list if it exists in the map)